ner/src/main/java/edu/illinois/cs/cogcomp/ner/BenchmarkOutputParser.java

```diff
@@ -0,0 +1,356 @@
+    
+    /** get the F1 score for phrase only.  */
+    static Pattern phraselevelpattern = Pattern.compile(
+        "Phrase-level F1 on the dataset:[^\n]*[\n]+"
+        + "[\\s]*Level 1:[\\s]*([\\d\\.]*)[^\\n]*[\\n]+"
+        + "[\\s]*Level 2:[\\s]*([\\d\\.]*)");
+    
+    /**
+     * report an error.
+     * @param message the error message
+     */
+    static private void reportError(String message) {
+        System.err.println(message);
+        System.out.println("java edu.illinois.cs.cogcomp.nerBenchmarkOutputParser -single <results_file_name>");
+        System.exit(1);
+    }
+    
+    /**
+     * Parse the arguments, look for the file name, and possible "-single" flag.
+     * @param args
+     */
+    static private void parseArgs(String[] args) {
+        for (int i = 0; i < args.length; i++) {
+            if (args[i].equals("-single")) {
+                single = true;
+            } else {
+                resultsfile = new File(args[i]);
+                if (!resultsfile.exists()) {
+                    reportError("The directory name \""+args[i]+"\" did not exist, we expect directories with benchmark runs within. ");
+                }
+                if (!resultsfile.isDirectory()) {
+                    reportError("The file \""+args[i]+"\" was not a directory, we expected a directory with benchmark runs within. ");
+                }
+            }
+        }
+        if (resultsfile == null)
+            reportError("You must specify a directory where the results are found.");
+    }
+    
+    /**
+     * get the learning rates and thicknesses from the file name.
+     * @param file the file containing the results to parse, filename includes the params.
+     * @return the set of parameters for the learner.
+     */
+    static Parameters parseFilename(File file) {
+        String name = file.getName();
+        float l1lr, l2lr, l1t, l2t;
+        int s = name.indexOf("L1r");
+        int e = name.indexOf("-t");
+        String sub = name.substring(s+3, e);
+        l1lr = Float.parseFloat(sub);
+        
+        s = e+2;
+        e = name.indexOf("+L2r");
+        sub = name.substring(s, e);
+        l1t = Float.parseFloat(sub);
+        
+        s = e+4;
+        e = name.indexOf("-t", e);
+        sub = name.substring(s, e);
+        l2lr = Float.parseFloat(sub);
+        
+        s = e+2;
+        sub = name.substring(s);
+        l2t = Float.parseFloat(sub);
+        return new Parameters(l1lr, l1t, l2lr, l2t);
+    }
+    
+    
+    /**
+     * This main method will take one required argument, idenfitying the file containing 
+     * the results. Optionally, "-single" may also be passed indicating it will extract
+     * the F1 value for single token values only.
+     * @param args
+     * @throws IOException 
+     */
+    public static void main1(String[] args) throws IOException {
+        /** get the F1 score for token only.  */
+        String fuck = "Token-level Acc Level1:\n" + 
```

Comment:
While guilty of the odd obscenity myself in comments, maybe rename this variable?

Reply:
hehehe. Whoops. that was a test method and should have been removed.